apis/clusters/v1beta1/cadence_webhook.go

```diff
@@ -255,6 +259,11 @@ func (cv *cadenceValidator) ValidateUpdate(ctx context.Context, old runtime.Obje
 		}
 	}
 
+	// ensuring if the cluster is ready for the spec updating
+	if (c.Status.CurrentClusterOperationStatus != models.NoOperation || c.Status.State != models.RunningStatus) && c.Generation != oldCluster.Generation {
```

Comment:
Code should be self documented, I don't think that we need such comments

Reply:
I discussed it with teammates and we decided to keep it as it is because it enhances the readability of the following check

Reply:
Strange to write comment for check if no operation or not running status, but ok

Reply:
This comment refers to spec update only, no the general k8s resource update